vsintegration/src/FSharp.Editor/Common/Extensions.fs

```diff
@@ -40,6 +42,47 @@ type Document with
                 languageServices.GetService<'T>()
                 |> Some
 
+type TextLine with
+
+    member this.ToFSharpTextLine() =
+        { new ITextLine with
+
+            member __.Length = this.Span.Length
+
+            member __.TextString = this.ToString()
```

Comment:
Don't like `TextString` - need to use `{ new System.Object() with` and override `ToString()` IMO.

Reply:
I can't see where this is used?  I think the `.ToString()` call in `getXmlDocablesOfSourceText` should be `.TextString`?  If so, then maybe you can remove `ITextLine` and `ITextLineCollection` and simply have a `GetLines()` on `ISourceText` that returns a `string[]`, since that's what `getXmlDocablesOfSourceText` is ultimately doing.   I'll poke around for other used of those though

```
                for i = 0 to sourceText.Lines.Count - 1 do
                    yield sourceText.Lines.[i].ToString()
```

Reply:
Originally I had something like this. I guess I don't mind either way.